### ISSUE TITLE
Speaker Diarization takes too long

### ISSUE THREAD

**Issue by thuthuts** [NVIDIA/NeMo#6674]
I have finetuned a speaker recognition model with the help of this tutorial : https://github.com/NVIDIA/NeMo/blob/main/tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb
After that I trained a MSDD Model for Speaker Diarization and replaced the pretrained titanet model with my finetuned Speaker Recognition Model. I followed this tutorial :https://github.com/NVIDIA/NeMo/blob/main/tutorials/speaker_tasks/Speaker_Diarization_Training.ipynb
I tried both, training with a frozen speaker embeddings model and joint training and noticed that the validation loss was much lower when using joint training. Nevertheless, the inference time for Speaker Diarization takes forever when extracting embeddings. How can I tackle that problem?

**Comment by gabitza-tech** [comment]
Hi, can you try removing the backprop gradients calculation when extracting embeddings in the nemo/collections/asr/models/clustering_diarizer.py? For me, it sped up the process by a little:

```python
 def _extract_embeddings(self, manifest_file: str, scale_idx: int, num_scales: int):
        """
        This method extracts speaker embeddings from segments passed through manifest_file
        Optionally you may save the intermediate speaker embeddings for debugging or any use. 
        """
        self._setup_spkr_test_data(manifest_file)
        self.embeddings = {}
        self._speaker_model = self._speaker_model.to(self._device)
        self._speaker_model.eval()
        self.time_stamps = {}
        
        all_embs = torch.empty([0])
        for test_batch in tqdm(self._speaker_model.test_dataloader(), desc=f'[{scale_idx+1}/{num_scales}] extract embeddings',leave=True):
            test_batch = [x.to(self._device) for x in test_batch]
            with autocast():
                with torch.no_grad():
                    _, embs = self._speaker_model.forward(input_signal=test_batch[0], input_signal_length=test_batch[1])
                emb_shape = embs.shape[-1]
                embs = embs.view(-1, emb_shape)
                all_embs = torch.cat((all_embs, embs.cpu().detach()), dim=0)
                
            del test_batch
```

**Comment by great-goblin** [comment]
Is your CUDA correctly configured?

`$nvcc --version`
